MekHQ/src/mekhq/campaign/unit/Unit.java

```diff
@@ -4989,11 +4991,24 @@ public int getMaintenanceTime() {
         return retVal * getMaintenanceMultiplier();
     }
 
-    public void incrementDaysSinceMaintenance(boolean maintained, int astechs) {
-        daysSinceMaintenance++;
-        astechDaysMaintained += astechs;
+    public void incrementDaysSinceMaintenance(Campaign campaign, boolean maintained, int astechs) {
+        List<Mission> activeMissions = campaign.getActiveMissions(false);
+        double timeIncrease = 0.25;
+
+        for (Mission mission: activeMissions) {
+            if (mission instanceof AtBContract) {
+                if (((AtBContract) mission).getContractType().isGarrisonDuty()) {
+                    continue;
+                }
+            }
+
+            timeIncrease = 1;
```

Comment:
If we're setting this to a fixed value, we can break out of the loop after the first time we set it.